internal/action/command.go

```diff
@@ -354,13 +354,23 @@ func reloadRuntime(reloadPlugins bool) {
 		config.InitPlugins()
 	}
 
+	oldParsedSettings := config.ParsedSettings()
+
 	err := config.ReadSettings()
 	if err != nil {
 		screen.TermMessage(err)
 	}
-	err = config.InitGlobalSettings()
-	if err != nil {
-		screen.TermMessage(err)
+
+	parsedSettings := config.ParsedSettings()
+	defaultGlobals := config.DefaultGlobalSettings()
+	for k := range defaultGlobals {
+		if _, ok := parsedSettings[k]; ok {
+			SetGlobalOptionNative(k, parsedSettings[k])
+			continue
+		}
+		if _, ok := oldParsedSettings[k]; ok {
+			SetGlobalOptionNative(k, defaultGlobals[k])
+		}
```

Comment:
1. We should do this for local settings as well.
2. We should do this for local settings not just in `reload` but also in `set filetype`.

For example, with this in `settings.json`:

```
    "ft:c": {
        "ruler": false
    },
```

open `foo.go`, run `set filetype c` -> the ruler disappears (as expected)

but then run `set filetype go` -> the ruler doesn't appear again.